### PR TITLE
F extract delete statement methods

### PR DESCRIPTION
**Pull request by martendemos**
Reduce complexity of StatementService by reducing size by extracting methods into own servcies.

Also eliminate useages of StatementService::deleteStatement by replacing usages with StatementDeleter::deleteStatementObject.

### How to review/test
Deletion of statements should be still possible.

- [x] Tests updated/created

### CODE REVIEW

File: demosplan/DemosPlanAssessmentTableBundle/Logic/AbstractRpcStatementBulkAction.php [hunk 1]
```diff
@@ -99,6 +100,7 @@ abstract class AbstractRpcStatementBulkAction implements RpcMethodSolverInterfac
      * @var TransactionService
      */
     private $transactionService;
+    protected StatementDeleter $statementDeleter;
```
Comment (dresslerdemos) [inline demosplan/DemosPlanAssessmentTableBundle/Logic/AbstractRpcStatementBulkAction.php:103]: Using `protected readonly StatementDeleter $statementDeleter` in the constructor would be more concise.

File: demosplan/DemosPlanAssessmentTableBundle/Logic/AssessmentTableServiceStorage.php [hunk 2]
```diff
@@ -115,6 +116,7 @@ class AssessmentTableServiceStorage
      * @var CurrentUserService
      */
     private $currentUser;
+    private StatementDeleter $statementDeleter;
```
Comment (dresslerdemos) [inline demosplan/DemosPlanAssessmentTableBundle/Logic/AssessmentTableServiceStorage.php:119]: Using `private readonly StatementDeleter $statementDeleter` in the constructor would be more concise.

**Review by dresslerdemos — DISMISSED** [review]
Implementation acceptable. Didn't review changes to tests.

**Comment by mussbach** [comment]
@martendemos what about this?

**Comment by martendemos** [comment]
> @martendemos what about this?

This is a kind of clean up and has no functional requirement for release. 
I will change the target branch to main.

File: demosplan/DemosPlanAssessmentTableBundle/Logic/AssessmentTableServiceStorage.php (hunk 2 above)
Comment (martendemos) [reply demosplan/DemosPlanAssessmentTableBundle/Logic/AssessmentTableServiceStorage.php]: a6b381f0b78add7ef5b84ba014ad19745b646ac0

File: demosplan/DemosPlanAssessmentTableBundle/Logic/AbstractRpcStatementBulkAction.php (hunk 1 above)
Comment (martendemos) [reply demosplan/DemosPlanAssessmentTableBundle/Logic/AbstractRpcStatementBulkAction.php]: a6b381f0b78add7ef5b84ba014ad19745b646ac0

Comment (dresslerdemos) [reply demosplan/DemosPlanAssessmentTableBundle/Logic/AbstractRpcStatementBulkAction.php]: What I meant was the usage of [constructor property promotion](https://www.php.net/manual/en/language.oop5.decon.php#language.oop5.decon.constructor.promotion) in conjunction with [readonly properties](https://www.php.net/manual/en/language.oop5.properties.php#language.oop5.properties.readonly-properties).

**Comment by mussbach** [comment]
@dresslerdemos could you please re-review and @martendemos could you please resolve the merge conflicts? It would be a pity to let this gem go

**Review by dresslerdemos — APPROVED** [review]
This is fine. But please take note of https://github.com/demos-europe/demosplan-core/pull/1164#discussion_r1176160555

**Pull request merged**